src/components/directory-content.js

```diff
@@ -865,6 +974,24 @@ const DirectoryContent = () => {
                                     )}
                                 </>
                             )}
+                            {allowsCopyFilterToScript() && (
```

Comment:
We should add a function to build MenuItems because we mix allowance checks and types checks to show the good menu which is not clever.

Reply:
refactoring this is not really the goal of my PR, I followed the way it is made. should probably make another pr if we want to change this

Reply:
true